stripped/colabs/Detekcja_sezonowosci_stripped.py

```diff
@@ -0,0 +1,915 @@
+    """
+    Funkcja, która trenuje model SARIMAX.
+    Parametry:
+    - train_df (Pandas DataFrame): zbiór treningowy,
+    - attribute (str): nazwa kolumny będąca zmienną zależną (dependent variable),
+    - year_split (int): rok rozdzielający zbiór treningowy i testowy,
+    - p, d, q (int): parametry funkcji SARIMAX, argumentu order,
+    - P, D, Q, s (int): parametry funkcji SARIMAX, argumantu seasonal_order,
+    - enforce_stat (bool): parametr determinujący argument enforce_stationarity funkcji SARIMAX.
+    """
+    sarima = SARIMAX(train_df[attribute],
+                order=(p,d,q),
+                seasonal_order=(P,D,Q,s),
+                trend="ct",
+                enforce_stationarity=enforce_stat).fit()
+    return sarima
+  sarima = train_model(train_data, attribute, p, d, q, P, D, Q, s, enforce_stat)
+  # Test modelu i wyświetlenie wyników
+  def model_predict(model, year_split: int):
+    """
+    Funkcja dokonuje predykcji za pomocą wczytanego modelu począwszy od pierwszego stycznia roku year_split do 2023-09-01.
+    - model: model za pomocą, którego dokonujemy predykcji,
+    - year_split (int): rok rozdzielający zbiór treningowy i testowy.
+    """
+    df_pred = model.predict(start= str(year_split) + "-01-01", end="2023-09-01", steps=len(test_data))
+    return df_pred
+  df_pred = model_predict(sarima, year_split)
+  # Wyświetlenie metryk
+  def print_mean_squerd_error_and_r2_score(test_df: pd.DataFrame, test_pred: pd.DataFrame, attribute: str) -> None:
+    """
+    Funkcja wyświetlająca błąd średniokwadratowy oraz współczynnik determinacji.
+    - test_df (Pandas DataFrame): zbiór testowy,
+    - test_pred (Pandas DataFrame): zbiór zawierające predykcje modelu,
+    - attribute (str): nazwa kolumny będąca zmienną zależną (dependent variable).
+    """
+    print(f"Mean squared error: {round(mean_squared_error(test_df[attribute], test_pred),3)}")
+    print(f"Coefficient of determination: {round(r2_score(test_df[attribute], test_pred),3)}")
+  print_mean_squerd_error_and_r2_score(test_data, df_pred, attribute)
+  # Tworzenie wykresu predykcji
+  def plot_predictions(test_data: pd.DataFrame, test_pred: pd.DataFrame, attribute: str) -> None:
+    """
+    Funkcja tworzy wykres predykcji modelu i prawdziwych wartości.
+    Parametry:
+    - test_data (Pandas DataFrame): zbiór testowy, który zawiera badany atrybut,
+    - test_pred (Pandas DataFrame): zbiór zawierające predykcje modelu,
+    - attribute (str): nazwa kolumny będąca zmienną zależną (dependent variable).
+    """
+    plt.figure(figsize=(7, 4), dpi=100)
+    test_data[attribute].plot(label=f'Prawdziwa wartość {attribute}', color='green')
+    plt.plot(test_data[attribute].index, test_pred, label='Predykcja modelu', color='purple')
+    plt.title(f'Model regresji liniowej dla {attribute}')
+    plt.xlabel('Data')
+    plt.ylabel(attribute)
+    plt.legend(loc='upper left')
+    plt.show()
```

Comment:
Niepotrzebne osadzenie we wnetrzu funkcji

Reply:
Poprawione